module/validation/receipt_validator.go

```diff
@@ -160,64 +161,203 @@ func (v *receiptValidator) Validate(receipts []*flow.ExecutionReceipt) error {
 	// reference each other.
 	// ATTENTION: Here we assume that ER is valid, this lookup can return a result which is actually invalid.
 	// Eventually invalid result will be detected and fail the whole validation.
-	previousResult := func(executionResult *flow.ExecutionResult) (*flow.ExecutionResult, error) {
-		prevResult, found := payloadExecutionResults[executionResult.PreviousResultID]
+	fetchResult := func(previousResultID flow.Identifier) (*flow.ExecutionResult, error) {
+		prevResult, found := payloadExecutionResults[previousResultID]
 		if found {
 			return prevResult, nil
 		}
 
-		return v.previousResult(executionResult)
+		return v.fetchResult(previousResultID)
 	}
 
 	for i, r := range receipts {
```

Comment:
why would the `receiptValidator` validate multiple receipts, if it's not validating the block payload.